ff_dir.c

```diff
@@ -2903,11 +2910,12 @@ static const uint8_t forbiddenChrs[] =
         {
             if( *pcName == forbiddenChrs[ index ] )
             {
-                *pcName = '_';
-                break;
+                xReturn = pdFALSE;
```

Comment:
Might want to look at: [Error codes](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-FAT/blob/bfb03575759a18e54f1ae3c1dbcb464741463589/include/ff_error.h#L174) and select a more appropriate one.
```suggestion
                xReturn = FF_ERR_FILE_COULD_NOT_CREATE_DIRENT;
```

Reply:
@phelter wrote:
> Might want to look at: [Error codes](https://github.com/FreeRTOS/Lab-Project-FreeRTOS-FAT/blob/bfb03575759a18e54f1ae3c1dbcb464741463589/include/ff_error.h#L174) and select a more appropriate one.

I do not see any reason to use an `FF_Error_t` here because the function is only checking the validity of a string. No disk access is needed.